protostar/testing/cheatcodes/reflect/cairo_struct.py

```diff
@@ -1,45 +1,31 @@
-from typing import Any, List, OrderedDict, Tuple, Type, Union
+from collections import OrderedDict
+from typing import Any, Union, cast
 
 from starkware.cairo.lang.vm.relocatable import RelocatableValue
 
-from protostar.starknet import HintLocal, SimpleBreakingReportedException
-
-
-class CairoStructException(SimpleBreakingReportedException):
-    pass
+from protostar.starknet import HintLocal
 
 
 class CairoStruct:
-    def __init__(self, *args, **kwargs) -> None:
-        if len(args) > 0:  # / syntax not available in py3.7
-            raise CairoStructException(
-                "CairoStruct constructor takes only keyword arguments."
-            )
-
+    def __init__(self, **kwargs) -> None:
         self._ordered_dict: OrderedDict[str, CairoValueType] = OrderedDict()
 
         for key, value in kwargs.items():
-            if not isinstance(value, VALID_CAIRO_TYPES):
-                raise CairoStructException(
-                    f'"{type(value).__name__}" is not a valid CairoType.'
-                )
-            self._ordered_dict[key] = value
+            self._set(key, value)
 
     def __getattr__(self, name: str) -> "CairoValueType":
         if name in self._ordered_dict:
             return self._ordered_dict[name]
-        raise CairoStructException(f'"{name}" is not a member of this CairoStruct.')
+        raise KeyError(f"'{name}' is not a member of this CairoStruct.")
 
     def __setattr__(self, name: str, value: Any) -> None:
         if name != "_ordered_dict":
-            raise CairoStructException("CairoStruct is immutable.")
+            raise ValueError("CairoStruct is immutable.")
```

Comment:
These exceptions will be thrown with a stacktrace as an unexpected exception.

Reply:
Discussed this in office.